sql/opt_range.h

```diff
@@ -1651,6 +1651,39 @@ class SQL_SELECT :public Sql_alloc {
 };
 
 
+class SQL_SELECT_auto: public SQL_SELECT
```

Comment:
I guess you wanted RAII for `SQL_SELECT*`. Something like std::unique_ptr. I like it too very much.

But this class has nothing to do with AS OF and our big patch is not a place to introduce such classes. If your really want something like this I suggest to make a PR to MariaDB and then use it in AS OF code.

Apart from that class looks very strange. It both inherits from `SQL_SELECT` and has a pointer to it as a member, it's base class has a non-virtual destructor. And I use this class like that and this will be incorrect, right? Base class destructor won't be called and a memory leak will happen.
```c++
SQL_SELECT_auto s;
s.quick = p;
```

Reply:
I don't see much reason to create separate PRs for such trifle subjects.

This seems to be the typo: inheritance from `SQL_SELECT`.